src/Message/ScanFileForVirus.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace App\Message;
+
+/**
+ * Message for virus-scanning files.
+ */
+class ScanFileForVirus
+{
+    /**
+     * The stream to be scanned.
+     *
+     * @var resource
+     */
+    protected $stream;
+
+    /**
+     * The UDI of the associated stream, for identification purposes.
+     *
+     * @var string udi
+     */
+    protected $udi;
+
+    /**
+     * The ID of the file assocated with the stream, for identification purposes.
+     *
+     * @var int id
+     */
+    protected $id;
+
+    /**
+     * Constructor.
+     *
+     * @param array    The stream and attributes describing the associated file for who's stream is to be scanned.
+     * @param string   The UDI associted with the stream.
+     * @param int      The fileId associated with the stream.
+     */
+    public function __construct(array $stream, string $udi, int $id)
```

Comment:
Add :void for the PSR12 gods...

Reply:
Constructors, destructors, and the clone operator cannot declare return types though.
https://wiki.php.net/rfc/return_types